emukit/bayesian_optimization/acquisitions/expected_improvement_per_cost.py

```diff
@@ -0,0 +1,61 @@
+from typing import Tuple, Union
+
+import numpy as np
+
+from ...core.interfaces import IModel, IDifferentiable
+from ...core.acquisition import Acquisition
+from emukit.bayesian_optimization.acquisitions import ExpectedImprovement
+
+
+class ExpectedImprovementPerCost(Acquisition):
+
+    def __init__(self, model: Union[IModel, IDifferentiable], cost_model: Union[IModel, IDifferentiable],
+                 jitter: np.float64 = np.float64(0))-> None:
+        """
+        This acquisition computes for a given input the improvement over the current best observed value in
+        expectation. For more information see:
+
+        Efficient Global Optimization of Expensive Black-Box Functions
+        Jones, Donald R. and Schonlau, Matthias and Welch, William J.
+        Journal of Global Optimization
+
+        :param model: model that is used to compute the improvement.
+        :param jitter: parameter to encourage extra exploration.
+        """
+
+        self.model = model
+        self.cost_model = cost_model
+        self.jitter = jitter
+
+        self.ei = ExpectedImprovement(model, jitter)
+
+    def evaluate(self, x: np.ndarray) -> np.ndarray:
+        """
+        Computes the Expected Improvement.
+
+        :param x: points where the acquisition is evaluated.
+        """
+        improvement = self.ei.evaluate(x)
+
+        mean, _ = self.cost_model.predict(x)
+
+        return improvement / mean
```

Comment:
does the cost model make sure that cost mean is not zero?

Reply:
good point, to prevent that I usually model the log cost. We could either model the log cost as default or just add a very small value to the cost for numerical stability. What do you think?

Reply:
I like the idea of modelling log cost more because it enforces positivity. We might also need to add a small positive value in case the cost is extremely small.

Reply:
the transformation should happen inside the cost model though. There might be other transformations which are suitable especially when cost_model is a distribution itself. for example analytic approximations to xi-square distributions are easier. 

I agree that we might need to add a small positive value, possibly even a bound, This is because we might run into trouble with the explorative behavior of the cost-sensitive acquisition if the cost is not strictly bounded away from zero (bounded below by zero might not be enough).